pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -252,7 +252,7 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 
 	var passthroughModSecurityRuleSet bool
 	var modSecurityRuleSet *corev1.ConfigMap
-	if r.isWAFEnabled(&instance.Spec) {
+	if r.isWAFEnabled(&instance.Spec) || r.isSidecarInjectionEnabled(&instance.Spec) {
```

Comment:
This seems to suggest that `SidecarInjection` is an alternative to WAF, whereas I had been thinking of `SidecarInjection` as a config knob on WAF (and other L7 features).

e.g., 

```yaml
spec:
  WebApplicationFirewall: Enabled
  SidecarInjection: Enabled
```

^ This seems valid to me. I would expect WAF to be enabled, and implemented using a sidecar.

```yaml
spec:
  WebApplicationFirewall: Enabled
  SidecarInjection: Disabled
```

^ This also seems valid to me. I would expect WAF to be enabled, but implemented using a per-host proxy.

```yaml
spec:
  WebApplicationFirewall: Disabled
  SidecarInjection: Enabled
```

^ This feels like an invalid configuration to me - from a UX perspective, injecting a sidecar doesn't do anything on its own unless the sidecar is configured to perform WAF / ALP / L7 logging / etc. as well.

Reply:
@radixo OK, so I am sort of following that - sounds like it is valid to have both daemonset and sidecar injection enabled at the same time.

> because it will depend on the pod annotations if WAF is enabled for that por or not, 

Right, so there will be a new `applicationlayer.projectcalico.org/waf` annotation that will control whether or not WAF is enabled for sidecar-based pods (and similar for other L7 features)

Does that mean it is viable to have _only_ sidecar injection enabled (with no DaemonSet)?

e.g.,

```yaml
spec:
  sidecarInjection: Enabled
  webApplicationFirewall: Disabled
```

If so, this is a confusing API, since the user is saying "Disable WAF" but what is actually being implemented is "Enable WAF per-pod instead of per-host"

Not sure how we could make that better, but figured it should be called out. Perhaps we just need detailed API comments / documentation to explain that.

Reply:
I think we leave this as-is for now.

I think we should consider cleaning this API up in a future version though. As a strawman, something like this seems more accurate:

```yaml
hostEnforcement:
  webApplicationFirewall: Disabled
  l7Logging: Enabled
sidecarEnforcement: Enabled
```

Reply:
Yeah, it's confusing... totally agree. I created this issue for Runtime Security backlog
https://tigera.atlassian.net/browse/RS-2061